### DISCUSSION TITLE
Is it possible to generate a static binary of OpenROAD?

### DISCUSSION THREAD

**Discussion by TiagoAFontana** [The-OpenROAD-Project/OpenROAD#1951] [Q&A]
Hi all, I am trying to execute the OpenROAD in a cluster machine. However, in this machine, I am not able to install the dependencies to compile the OpenROAD or even use the docker image to run the OpenROAD.

I was wondering if it is possible to set a cmake flag to generate a static binary (without any dependencies) for the OpenROAD project. With this, I will be able to run this binary on every machine that I need.

Does someone know how to do this?

**Comment by vijayank88** [comment]
@vvbandeira  Any suggestion?

**Reply by vvbandeira** [reply]
I do not think we support this at the moment, but you can certainly try to compile only with static libraries. A suggestion would be to first disable optional features like Python and GUI support (`-DBUILD_PYTHON=OFF -DENABLE_GUI=OFF`), since you are running on a cluster I think this should be fine. This would likely make it easier.

Another alternative you could try is to use a program like `guix` or `conda` that do not require root to install packages. Then you can use them to install the OpenROAD dependencies.

**Comment by QuantamHD** [comment]
Bazel rules hdl supports fully static builds of openroad.

Just download bazel, git clone https://github.com/hdl/bazel_rules_hdl

And run bazel build @org_theopenroadproject//:openroad in the git repo. 

It also statically builds the entire dependency chain of openroad from source.
